daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -270,14 +277,17 @@ func (l *logStream) collectBatch() {
 // publishBatch calls PutLogEvents for a given set of InputLogEvents,
 // accounting for sequencing requirements (each request must reference the
 // sequence token returned by the previous request).
-func (l *logStream) publishBatch(events []*cloudwatchlogs.InputLogEvent) {
+func (l *logStream) publishBatch(events []wrappedEvent) {
 	if len(events) == 0 {
 		return
 	}
 
+	// events in a batch must be sorted by timestamp
+	// see http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
 	sort.Sort(byTimestamp(events))
+	cwEvents := getCWEvents(events)
```

Comment:
As just discussed on Slack; I overlooked that the timestamps are not *generated* in `collectBatch`, but taken from the `msg`, so they *can* arrive in `collectBatch` in the wrong order.